index.js

```diff
@@ -502,12 +502,16 @@ Router.prototype.use = function use(handler) {
  * and middleware to routes.
  *
  * @param {string} path
+ * @param {string} name (optional)
  * @return {Route}
  * @public
  */
 
-Router.prototype.route = function route(path) {
-  var route = new Route(path)
+Router.prototype.route = function route(path, name) {
+  if(arguments.length > 1 && this.findRoute(name) != null) {
```

Comment:
simpler to just check `if (name && ...`.  IMO that is also more explicit about the intention of this code.